### PR TITLE
PACA-422: Add sequence numbers and timestamps

### PR DESCRIPTION
**Pull request by KyRobbins**
**JIRA Ticket:**
[PACA-422](https://jira.cms.gov/browse/PACA-422)

**User Story or Bug Summary:**

---

### What Does This PR Do?

RDA Claims need to include sequence numbers to allow for proper ingestion.  Additionally, the generated files will be marked with their contained sequence range to allow for proper identification of data sets that have not yet been consumed.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java [hunk 1]
```diff
@@ -96,6 +98,9 @@ public MessageOrBuilder transform(Parser.Data<String> data) {
       addDiagnosisCodes(claimBuilder, data, claimBuilder.getIdrClmHdIcn());
 
       return McsClaimChange.newBuilder()
+          .setTimestamp(
+              Timestamp.newBuilder().setSeconds(System.currentTimeMillis() / 1000).build())
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java:102]: `Instant.now().getEpochSecond()` might be more idiomatic than the division.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/io/NdJsonSink.java [hunk 2]
```diff
@@ -28,5 +43,33 @@ public void write(MessageOrBuilder messageOrBuilder) {
...
+  }
+
+  private Long getSequenceNumber(MessageOrBuilder messageOrBuilder) {
+    Long seqNumber = -1L;
+
+    for (Map.Entry<Descriptors.FieldDescriptor, Object> entry :
+        messageOrBuilder.getAllFields().entrySet()) {
+      if (entry.getKey().getFullName().endsWith("seq")) {
+        seqNumber = (Long) entry.getValue();
+      }
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/io/NdJsonSink.java:70]: Why `endsWith`?

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/io/NdJsonSink.java [hunk 3]
```diff
@@ -28,5 +38,19 @@ public void write(MessageOrBuilder messageOrBuilder) {
   @Override
   public void close() throws IOException {
     writer.close();
+
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/io/NdJsonSink.java:41]: Could WrappedCounter just track its min/max values so you don't need this deep knowledge of the message objects?

**Review by brianburton — APPROVED** [review]
Looks good!

**Review by ericbuckley — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**